tmt/utils.py

```diff
@@ -1077,6 +1112,26 @@ def default_branch(repository, remote='origin'):
         return ref.read().strip().split('/')[-1]
 
 
+def parse_dotenv(content: str) -> Dict[str, str]:
+    try:
+        return dict([line.split("=")
+                    for line in shlex.split(content, comments=True)])
+        # return dict(map(split_func, shlex.split(content, comments=True)))
+    except ValueError:
+        raise ValueError(f"Can't extract variables from {content}. "
+                         f"Ensure it has proper format (i.e. A=B)")
+
+
+def parse_yaml(content: str) -> Dict[str, str]:
+    yaml_as_dict = yaml.safe_load(content)
+    if any(isinstance(val, dict) for val in yaml_as_dict.values()):
+        raise GeneralError(
+            "Can't set the environment from the nested yaml config. The "
+            "config should be just key, value pairs."
+            )
+    return {k: str(v) for k, v in yaml_as_dict.items()}
+
```

Comment:
Instead of introducing new functions would it make sense to reuse `environment_to_dict()` and `yaml_to_dict()` which are already used for similar purpose? I guess the error handling could be done under `environment_file_to_dict()` as well?

Reply:
envirnomenr_to_dict
---------------------------
1. I was relying upon that environment_to_dict will be deprecated (`@file` functionality)
2. envirnomenr_to_dict is too generic for dotenv file, from the other side implementation of parse_dotenv is simple and just serves the purpose of parsing dotenv files.

yaml_to_dict
-----------------------
1. yaml_to_dict is the opposite - too simple. for parsing yaml format as source of env var I need more functionality (check if it is flat , convert values to str). Her more suitable _add_file_vars, but its interface is hardly compatible with my design.

Reply:
I see, thanks for clarification.